airsonic-main/src/test/java/org/airsonic/player/service/search/SearchServiceSpecialGenreTestCase.java

```diff
@@ -0,0 +1,309 @@
+         * as tag values ​​of Genre.
+         * 
+         * Legacy can not search all these genres.
+         * (Strictly speaking, the genre field is not created at index creation.)
+         */
+
+        List<MusicFolder> folders = getTestMusicFolders();
+
+        Function<String, RandomSearchCriteria> simpleStringCriteria = s ->
+            new RandomSearchCriteria(Integer.MAX_VALUE, // count
+                    s, // genre,
+                    null, // fromYear
+                    null, // toYear
+                    folders // musicFolders
+            );
+
+        List<MediaFile> songs = searchService.getRandomSongs(simpleStringCriteria.apply("+"));
+        Assert.assertEquals(0, songs.size());
+
+        songs = searchService.getRandomSongs(simpleStringCriteria.apply("-"));
+        Assert.assertEquals(0, songs.size());
+
+        songs = searchService.getRandomSongs(simpleStringCriteria.apply("&&"));
+        Assert.assertEquals(0, songs.size());
+
+        songs = searchService.getRandomSongs(simpleStringCriteria.apply("+"));
+        Assert.assertEquals(0, songs.size());
+
+        songs = searchService.getRandomSongs(simpleStringCriteria.apply("||"));
+        Assert.assertEquals(0, songs.size());
+
+        songs = searchService.getRandomSongs(simpleStringCriteria.apply(" ("));// space & bracket
+        Assert.assertEquals(0, songs.size());
+
+        songs = searchService.getRandomSongs(simpleStringCriteria.apply(")"));
+        Assert.assertEquals(0, songs.size());
+
+        songs = searchService.getRandomSongs(simpleStringCriteria.apply("{}"));
+        Assert.assertEquals(0, songs.size());
+
+        songs = searchService.getRandomSongs(simpleStringCriteria.apply("[]"));
+        Assert.assertEquals(0, songs.size());
+
+        songs = searchService.getRandomSongs(simpleStringCriteria.apply("^"));
+        Assert.assertEquals(0, songs.size());
+
+        songs = searchService.getRandomSongs(simpleStringCriteria.apply("\""));
+        Assert.assertEquals(0, songs.size());
+
+        songs = searchService.getRandomSongs(simpleStringCriteria.apply("~"));
+        Assert.assertEquals(0, songs.size());
+
+        songs = searchService.getRandomSongs(simpleStringCriteria.apply("*"));
+        Assert.assertEquals(0, songs.size());
+
+        songs = searchService.getRandomSongs(simpleStringCriteria.apply("?"));
+        Assert.assertEquals(0, songs.size());
+
+        songs = searchService.getRandomSongs(simpleStringCriteria.apply(":"));
+        Assert.assertEquals(0, songs.size());
+
+        songs = searchService.getRandomSongs(simpleStringCriteria.apply("\\"));
+        Assert.assertEquals(0, songs.size());
+
+        songs = searchService.getRandomSongs(simpleStringCriteria.apply("/"));
+        Assert.assertEquals(0, songs.size());
+
+    }
+
+    @Test
+    public void testBrackets() {
+        
+        /*
+         * Jaudiotagger applies special treatment to bracket (FILE17).
```

Comment:
Aren't comments going before the function in Java?

Reply:
I revised that according to your advice.